frontend/src/metabase-lib/filter.ts

```diff
@@ -556,6 +563,15 @@ function isStringOperator(
   return operators.includes(operator);
 }
 
+function getStringFilterOptions(
+  operator: ExpressionOperatorName,
+  options: ExpressionOptions,
+): StringFilterOptions {
+  const operators: ReadonlyArray<string> = STRING_FILTER_OPERATORS_WITH_OPTIONS;
+  const supportsOptions = operators.includes(operator);
+  return supportsOptions ? { "case-sensitive": false, ...options } : {};
```

Comment:
MLv2 breaks if we pass options for operators that do not support them. I think it's an implementation detail and we should handle it internally and not ask for consumers to know it